Bootloader/Src/firmware_update.c

```diff
@@ -242,10 +228,7 @@ FirmwareUpdate_communicationHandler(uint8_t* buf, uint32_t length) {
 
         case fwUpdateState_ERASE:
 
-            success = FlashAdapter_erase(firmware_size, s_flash_address);
-            if (success) {
-                FirmwareUpdate_ack();
-            }
+            success = FlashAdapter_erase(firmware_size, FLASH_FIRMWARE_ADDRESS);
 
             if (success) {
                 s_update_state = fwUpdateState_DOWNLOADING;

```

Comment:
We can rename this state to `fwUpdateState_DOWNLOADING_AND_FLASHING`, since it not only downloading anymore.  And state `fwUpdateState_FLASHING` can be deleted.